daffodil-runtime1/src/main/scala/org/apache/daffodil/runtime1/dsom/CompiledExpression1.scala

```diff
@@ -367,7 +367,7 @@ class DPathElementCompileInfo(
   /**
    * Cyclic objects require initialization
    */
-  override lazy val initialize: Unit = {
+  override def initialize(): Unit = {
```

Comment:
What is wrong with this as is? Changing this to def changes behavior from initializing only once. 
I think you have to do this to preserve exactly the behavior:
```
private lazy val init_ : Unit = {
   parents
   elementChildrenCompileInfo
}
override def initialize(): Unit = init_
```

Reply:
Scala 3 complains of overrding a def with a lazy value...I will implement your change above to make it the same functionality

 370 |  override lazy val initialize: Unit = {
[error]     |                    ^
[error]     |error overriding method initialize in class DPathCompileInfo of type (): Unit;
[error]     |  lazy value initialize of type Unit no longer has compatible type